feat(server): allow embed with listen (#25709)
neovim/neovim: runtime/doc/starting.txt

<		Then startup will continue without waiting for `nvim_ui_attach`.
		This is equivalent to: >
		    nvim --headless --cmd "call stdioopen({'rpc': v:true})"
<

> 		Embedders using the UI protocol and a socket connetction must

Reviewer: ```suggestion
		Embedders that use the UI protocol on a socket connection must
```

Author: done

> 		Embedders that use the UI protocol on a socket connetction must

Reviewer: ```suggestion
		Embedders that use the UI protocol on a socket connection must
```

Author: oops... thanks
